src/Features/CSharp/Portable/RemoveAsyncModifier/CSharpRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,106 @@
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpRemoveAsyncModifierCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(CS1998);
+
+        protected override bool IsAsyncSupportingFunctionSyntax(SyntaxNode node)
+            => node.IsAsyncSupportingFunctionSyntax();
+
+        protected override bool ShouldOfferFix(IMethodSymbol methodSymbol, KnownTypes knownTypes)
+        {
+            // For async void methods this fixer does the same as Make Method Synchronous so we don't need to offer both
+            if (methodSymbol.ReturnsVoid)
+            {
+                return false;
+            }
+
+            // IAsyncEnumerable iterators cannot be made non-async even if they don't have awaits
+            return !methodSymbol.ReturnType.OriginalDefinition.Equals(knownTypes._iAsyncEnumerableOfTTypeOpt);
+        }
+
+        protected override SyntaxNode GetLastChildOfBlock(SyntaxNode node)
+            => GetBlockBody(node).ChildNodes().Last();
+
+        protected override ControlFlowAnalysis AnalyzeControlFlow(SemanticModel semanticModel, SyntaxNode node)
+            => semanticModel.AnalyzeControlFlow(GetBlockBody(node));
+
+        private static SyntaxNode GetBlockBody(SyntaxNode node)
+            => node switch
+            {
+                MethodDeclarationSyntax method => method.Body,
+                LocalFunctionStatementSyntax localFunction => localFunction.Body,
+                AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.Block,
+                _ => throw ExceptionUtilities.Unreachable
+            };
+
+        protected override bool TryGetExpressionBody(SyntaxNode node, out SyntaxNode expression)
+        {
+            expression = node switch
+            {
+                // For methods and local functions ExpressionBody is an ArrowExpressionClauseSyntax so we pull the real expression out
+                MethodDeclarationSyntax method => method.ExpressionBody?.Expression,
+                LocalFunctionStatementSyntax localFunction => localFunction.ExpressionBody?.Expression,
+                AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.ExpressionBody,
+                _ => throw ExceptionUtilities.Unreachable
+            };
+
+            return expression != null;
+        }
+
+        protected override SyntaxNode ConvertToBlockBody(SyntaxNode node, SyntaxNode expressionBody, SyntaxEditor editor)
+        {
+            if (InitializeParameterHelpers.TryConvertExpressionBodyToStatement(expressionBody, SyntaxFactory.Token(SyntaxKind.SemicolonToken), false, out var statement))
```

Comment:
1. this is a shame we don't track the semicolon.  would be nice if we coudl grab that off of hte original construct and feed in her.
2. named param for 'false'.

Reply:
3. At this point, now that these helpers are used in a few places, this should mvoe out of InitializerParameterHelper.  We don't want features leaking their impl to others.  instead, this shoudl move to a utility location like ExpressionToBlockConverter or something of that nature.